### PR TITLE
Updated UPGRADE_NOTES for 0.25 update

### PR DESCRIPTION
**Pull request by jwoertink**
## Purpose
UPGRADE NOTES for the 0.25 update.

### CODE REVIEW

**Review by matthewmcgarvey — COMMENTED** [review]
The has_many through update is worth mentioning probably

**Comment by jwoertink** [comment]
Good catch. I'll hold off in case you or anyone else sees something I'm missing, but for sure that should go in.

File: UPGRADE_NOTES.md [hunk 1]
```diff
@@ -1,3 +1,128 @@
...
+
+- Run `shards update`
+
+### General updates
+
+- Update: all `Avram::Operation` to implement the new interface.
+  - Your main instance method should be called `run`
+  - The `run` method should return just the value you need. No more `yield self, thing` / `yield self, nil`.
+  - Call the operation with `MyOperation.run(params)` instead of `MyOperation.new(params).submit`
+  - The `MyOperation.run` method class method takes a block that yields the operation, and your return value. Similar to `SaveOperation`.
```
Comment (hanneskaeufler) [inline UPGRADE_NOTES.md:32]:
```suggestion
  - The `MyOperation.run` class method takes a block that yields the operation, and your return value. Similar to `SaveOperation`.
```

**Comment by jwoertink** [comment]
We should also mention https://github.com/luckyframework/lucky/discussions/1359#discussioncomment-216431

**Comment by matthewmcgarvey** [comment]
There will be places where certain query association methods won't work because of the `belongs_to` fixes. They just need to be singularized where they were plural. When I was updating my app, I was disappointed because crystal didn't offer a "did you mean xxx" in the error message so it could be confusing.

**Review by matthewmcgarvey — APPROVED** [review]

**Pull request merged**